9.2/run-postgresql.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash -e
+
+# Be paranoid and stricter than we should be.
+psql_identifier_regex='^[a-zA-Z_][a-zA-Z0-9_]*$'
+psql_password_regex='^[a-zA-Z0-9_~!@#$%^&*()-=<>,.?;:|]+$'
+
+function usage {
+	echo "You must specify following environment variables:"
+	echo "  \$POSTGRESQL_USER (regex: '$psql_identifier_regex')"
+	echo "  \$POSTGRESQL_PASSWORD (regex: '$psql_password_regex')"
+	echo "  \$POSTGRESQL_DATABASE (regex: '$psql_identifier_regex')"
+	echo "Optional:"
+	echo "  \$POSTGRESQL_ADMIN_PASSWORD (regex: '$psql_password_regex')"
+	exit 1
+}
+
+function valid_psql_identifier {
+	local var="$1" ; shift
+	[[ "${var}" =~ $psql_identifier_regex ]]
+}
+
+function valid_psql_password {
+	local var="$1" ; shift
+	[[ "${var}" =~ $psql_password_regex ]]
+}
+
+valid_psql_identifier "$POSTGRESQL_USER"     || usage
+valid_psql_password   "$POSTGRESQL_PASSWORD" || usage
+valid_psql_identifier "$POSTGRESQL_DATABASE" || usage
+
+# Make sure env variables don't propagate to mysqld process.
+postgresql_user="$POSTGRESQL_USER" ; unset POSTGRESQL_USER
+postgresql_pass="$POSTGRESQL_PASSWORD" ; unset POSTGRESQL_PASSWORD
+postgresql_db="$POSTGRESQL_DATABASE" ; unset POSTGRESQL_DATABASE
+
+# Admin password.
+if [ "$POSTGRESQL_ADMIN_PASSWORD" ]; then
+	valid_psql_password "$POSTGRESQL_ADMIN_PASSWORD" || usage
+	admin_pass="$POSTGRESQL_ADMIN_PASSWORD"
+fi
+unset POSTGRESQL_ADMIN_PASSWORD
+
+# SCL in CentOS/RHEL 7 doesn't support --exec, we need to do it ourselves
+# The '|| exit 1' is here so -e doesn't propagate into scl_source.
+source scl_source enable postgresql92 || exit 1
+
+function postgres_run {
+	runuser -u postgres -- "$@"
+}
+
+if [ ! -f "$PGDATA/postgresql.conf" ]; then
+
+	chown postgres.postgres "$PGDATA"
```

Comment:
A note about USER. We run this as root, but only thing that really requires root privileges is this chown. I would love to run the whole entrypoint as user postgres. However, we would have to make sure the volume is owned by the postgres user (uid 26).

Any ideas if this is possible?